crates/trie/sparse/src/state.rs

```diff
@@ -281,8 +281,11 @@ impl<F: BlindedProviderFactory> SparseStateTrie<F> {
 
     /// Reveal unknown trie paths from multiproof.
     /// NOTE: This method does not extensively validate the proof.
-    pub fn reveal_multiproof(&mut self, multiproof: MultiProof) -> SparseStateTrieResult<()> {
-        let MultiProof {
+    pub fn reveal_multiproof(
+        &mut self,
+        multiproof: DecodedMultiProof,
+    ) -> SparseStateTrieResult<()> {
```

Comment:
let's avoid changing the API and instead introduce `reveal_decoded_multiproof` which will be called by `reveal_multiproof`